modulus/launch/utils/checkpoint.py

```diff
@@ -345,21 +362,23 @@ def load_checkpoint(
             if isinstance(model, modulus.models.Module):
                 model.load(file_name)
             else:
-                model.load_state_dict(torch.load(file_name, map_location=device))
+                with fs.open(file_name, "rb") as fp:
+                    model.load_state_dict(torch.load(fp, map_location=device))
 
             checkpoint_logging.success(
                 f"Loaded model state dictionary {file_name} to device {device}"
             )
 
     # == Loading training checkpoint ==
     checkpoint_filename = _get_checkpoint_filename(path, index=epoch, model_type="pt")
-    if not Path(checkpoint_filename).is_file():
+    if not fs.exists(checkpoint_filename):
         checkpoint_logging.warning(
             "Could not find valid checkpoint file, skipping load"
         )
         return 0
 
-    checkpoint_dict = torch.load(checkpoint_filename, map_location=device)
+    with fs.open(checkpoint_filename, "rb") as fp:
```

Comment:
Wouldn't it be better to use `modulus.utils.filesystem._download_cached` here to also support ngc model registries?